[Enhance] Form -WIP proceeded to rendering

Zlash65 opened the pull request:
As of now TextView, EditText and CheckBox are rendered.

Note :- As of now Because the docData takes time to come from server, I had to make a holder array and wait for the response and then fill up the data. Need to do something about that. As soon as an Item is clicked the view is generated but the data hasn't come yet so added a ProgressBar to sugarcoat it.

Zlash65 committed:
Form WIP - phase 2

Zlash65 committed:
smart cast as edit or text view

Zlash65 committed:
form view generated along with data

revant commented:
https://github.com/agrinext/agrinext_mobile/blob/10aab09dcf05322f2f938ee2d2c8aa114954c034/app/src/main/java/org/agrinext/agrimobile/Activities/ListingActivity.kt#L186

Check doAsync - uiThread if it works. I am using it instead of settting a timeout
```
doAsync {
    asyncTask()
    uiThread {
        uiLoadThings()
    }
}
```

Zlash65 committed:
datepicker added, minor improvement

Zlash65 committed:
improvise and bifurcate code, additional functionality

Zlash65 commented:
Added "Add" functionality to open an empty form view.
Buttons added on ActionBar for Edit or Save.
Datepicker added for Datetime or Date fieldtype.
Cleaned up code and made a Form Utils Class to dump common code.
OH and just for fun added frappe colour to the actionBar, can change it back to green but I liked the Frappe-ized one more :P

revant commented on app/src/main/res/menu/list_view.xml:
```diff
@@ -16,4 +16,8 @@
         android:id="@+id/action_sort"/>
+
+    <item
+        android:title="Add"
```
difficult to translate 
use `@string/list_view_add`
